### PR TITLE
Pull in many to many association for cross-search from Hyku

### PR DESCRIPTION
**Pull request by mankind**
Pulling in Rob's Many to Many changes for shared-search from Hyku

### CODE REVIEW

File: db/migrate/20200929084240_add_parent_id_and_data_settings_column_to_account.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 class AddParentIdAndDataSettingsColumnToAccount < ActiveRecord::Migration[5.2]
   def change
-    add_reference :accounts, :parent, foreign_key: { to_table: :accounts }
+    # add_reference :accounts, :parent, foreign_key: { to_table: :accounts }
```
Comment (prdanelli) [inline db/migrate/20200929084240_add_parent_id_and_data_settings_column_to_account.rb:4]: This comment can be removed.

File: spec/models/account_cross_search_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe AccountCrossSearch, type: :model do
+  it do
+    is_expected.to belong_to(:full_account)
+      .class_name('Account')
+
+    is_expected.to belong_to(:search_account)
+      .class_name('Account')
```
Comment (prdanelli) [inline spec/models/account_cross_search_spec.rb:10]: Can these be on a single line to make it clearer what they are doing?

```
is_expected.to belong_to(:search_account).class_name('Account')
```

File: spec/requests/catalog_controller_spec.rb [hunk 3]
```diff
@@ -49,7 +62,8 @@
 
     context 'can fetch data from other tenants' do
       xit 'cross-search-tenant can fetch all record in child tenants' do
-        allow(blacklight_solr_repository).to receive(:build_connection).and_return(RSolr.connect(url: 'http://solr:8983/solr/hydra-cross-search-tenant'))
+        connection = RSolr.connect(url: 'http://solr:8983/solr/hydra-cross-search-tenant')
```
Comment (prdanelli) [inline spec/requests/catalog_controller_spec.rb:65]: Might be best to put the setup for this in a `let`.

File: db/migrate/20200929084240_add_parent_id_and_data_settings_column_to_account.rb (hunk 1 above)
Comment (mankind) [reply db/migrate/20200929084240_add_parent_id_and_data_settings_column_to_account.rb]: Thanks done.

File: spec/models/account_cross_search_spec.rb (hunk 2 above)
Comment (mankind) [reply spec/models/account_cross_search_spec.rb]: Thanks done.

File: spec/requests/catalog_controller_spec.rb (hunk 3 above)
Comment (mankind) [reply spec/requests/catalog_controller_spec.rb:65]: Since this test is **xit**, it is better to leave it till i revisit the whole test.  Thanks.

**Pull request merged**